app/src/org/commcare/android/resource/installers/OfflineUserRestoreAndroidInstaller.java

```diff
@@ -0,0 +1,90 @@
+package org.commcare.android.resource.installers;
+
+import org.commcare.CommCareApplication;
+import org.commcare.android.database.app.models.UserKeyRecord;
+import org.commcare.models.database.user.DatabaseUserOpenHelper;
+import org.commcare.resources.model.Resource;
+import org.commcare.resources.model.ResourceInitializationException;
+import org.commcare.resources.model.UnresolvedResourceException;
+import org.commcare.suite.model.OfflineUserRestore;
+import org.commcare.utils.AndroidCommCarePlatform;
+import org.javarosa.core.reference.InvalidReferenceException;
+import org.javarosa.core.reference.Reference;
+import org.javarosa.core.services.storage.EntityFilter;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Phillip Mates (pmates@dimagi.com)
+ * @author Aliza Stone (astone@dimagi.com)
+ */
+public class OfflineUserRestoreAndroidInstaller extends FileSystemInstaller {
+
+    @SuppressWarnings("unused")
+    public OfflineUserRestoreAndroidInstaller() {
+        // for externalization
+    }
+
+    public OfflineUserRestoreAndroidInstaller(String localDestination, String upgradeDestination) {
+        super(localDestination, upgradeDestination);
+    }
+
+    @Override
+    public boolean initialize(AndroidCommCarePlatform instance, boolean isUpgrade) throws ResourceInitializationException {
+        instance.registerDemoUserRestore(initDemoUserRestore());
+        return true;
+    }
+
+    private OfflineUserRestore initDemoUserRestore() throws ResourceInitializationException {
+        try {
+            return new OfflineUserRestore(localLocation);
+        } catch (UnfullfilledRequirementsException e) {
+            throw new ResourceInitializationException(e.getMessage());
```

Comment:
Agh, sorry for so much back and forth. Thanks for the super detailed review, really appreciate it. Should be g2g now